L2SIVacuum/POUs/Functions/Valves/FB_VCN_VAT590.TcPOU

```diff
@@ -200,46 +264,145 @@ IF tAction.Q THEN fbLogger(sMsg:='Valve commanded to fully open', eSevr:=TcEvent
 ]]></ST>
       </Implementation>
     </Action>
-    <Action Name="ACT_Persistent" Id="{e5168615-6ef3-4946-b318-5dac9816d744}">
+    <Action Name="ReadCoeParameters" Id="{dceab739-9779-4d3e-8c12-eb434a1eb90a}">
       <Implementation>
-        <ST><![CDATA[(*On first PLC pass, load the persistent value into the structrue variable*)
-IF (bRestorePersistentData) THEN
-	bRestorePersistentData	:= FALSE;
-	IF (fUpperLimit <> 0) THEN
-		stVcnVat590.fPosSetLimit := fUpperLimit;
-	END_IF;
-	IF (fPresMaxLim <> 0) THEN
-		stVcnVat590.fPresSetLimit := fPresMaxLim;
-	END_IF;
-	IF (fPresRbMaxLim <> 0) THEN
-		stVcnVat590.fPresRdbkLimit := fPresRbMaxLim;
-	END_IF;
-	
-	    
-END_IF]]></ST>
+        <ST><![CDATA[CASE nCaseReadCoe OF
+	0:
+		stVcnVat590.bCoeError := TRUE;
+		IF rtCoeReset.Q THEN
+			nCaseReadCoe := 1;
+			stVcnVat590.bCoeError := FALSE;
+			bFirstRead := TRUE; // Set bFirstRead after Coe interface reset
+		END_IF
+	1:
+		fbCoeRead.nIndex := cPressCtrl.nIndex;
+		fbCoeRead.nSubIndex := cPressCtrl.nSubIndex;
+		fbCoeRead.pDstBuf := ADR(stVcnVat590.stCoeParameters.ePIDCtrl);
```

Comment:
I notice ePIDControl is linked up to cPressCtrl. This is intentional, right?
I'm only asking because all the other cases matched up 1:1 precisely in their naming besides this one.

Reply:
Yes, so what happened here is that in the documentation this parameter is named as PRESSURE CONTROL ALGORITHM, but I named it as PID Controller, to me it was more logical. I can change it to PressCntrl if that will make more sense, let me know.

Reply:
No it's OK as is and it makes sense, I was just making sure there wasn't a typo. Sometimes things that look intentional are often typos and they get overlooked.